src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -6,6 +6,61 @@
 from proto.import_all_protos import *
 import software.thunderscope.common.common_widgets as common_widgets
 from software.thunderscope.constants import *
+from software.thunderscope.robot_diagnostics.motor_fault_view import MotorFaultView
+
+
+class BreakbeamLabel(QLabel):
+    """
+    Displays the current breakbeam status
+    Extension of a QLabel which displays a tooltip and updates the UI with the current status
+    """
+
+    BREAKBEAM_BORDER = "border: 2px solid black"
+
+    def __init__(self):
+        """
+        Constructs a breakbeam indicator and sets the UI to the default uninitialized state
+        """
+        super().__init__()
+        self.update_breakbeam_status(None)
+
+    def update_breakbeam_status(self, new_breakbeam_status):
+        """
+        Updates the current breakbeam status and refreshes the UI accordingly
+        :param new_breakbeam_status: the new breakbeam status
+        """
+        self.breakbeam_status = new_breakbeam_status
+
+        if self.breakbeam_status == None:
+            self.setStyleSheet(
+                f"background-color: transparent; {self.BREAKBEAM_BORDER}"
+            )
+        elif self.breakbeam_status:
+            self.setStyleSheet(
+                f"background-color: red; {self.BREAKBEAM_BORDER};" "border-color: red"
+            )
+        else:
+            self.setStyleSheet(
+                f"background-color: green; {self.BREAKBEAM_BORDER};"
+                "border-color: green"
+            )
+
+    def event(self, event):
+        """
+        Overridden event function which intercepts all events
+        On hover, displays a tooltip with the current breakbeam status
+        :param event: event to check
+        """
+        common_widgets.display_tooltip(
+            event,
+            "No Signal Yet"
+            if self.breakbeam_status == None
```

Comment:
To stay consistent with Arun's comment
```suggestion
            if self.breakbeam_status is None
```